app/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategy.java

```diff
@@ -0,0 +1,198 @@
+    }        
+    
+    public static class ScheduleCriteria implements Criteria {
+        private final Schedule schedule;
+        private final Integer minAppVersion;
+        private final Integer maxAppVersion;
+        private final Set<String> allOfGroups;
+        private final Set<String> noneOfGroups;
+        
+        @JsonCreator
+        private ScheduleCriteria(@JsonProperty("schedule") Schedule schedule, 
+                @JsonProperty("minAppVersion") Integer minAppVersion, 
+                @JsonProperty("maxAppVersion") Integer maxAppVersion, 
+                @JsonProperty("allOfGroups") Set<String> allOfGroups, 
+                @JsonProperty("noneOfGroups") Set<String> noneOfGroups) {
+            this.schedule = schedule;
+            this.minAppVersion = minAppVersion;
+            this.maxAppVersion = maxAppVersion;
+            this.allOfGroups = allOfGroups;
+            this.noneOfGroups = noneOfGroups;
+        }
+        public Schedule getSchedule() {
+            return schedule;
+        }
+        @Override
+        public Integer getMinAppVersion() {
+            return minAppVersion;
+        }
+        @Override
+        public Integer getMaxAppVersion() {
+            return maxAppVersion;
+        }
+        @Override
+        public Set<String> getAllOfGroups() {
+            return allOfGroups;
+        }
+        @Override
+        public Set<String> getNoneOfGroups() {
+            return noneOfGroups;
+        }
+        
+        public static class Builder {
+            private Schedule schedule;
+            private Integer minAppVersion;
+            private Integer maxAppVersion;
+            private Set<String> allOfGroups = Sets.newHashSet();
+            private Set<String> noneOfGroups = Sets.newHashSet();
+            
+            public Builder withSchedule(Schedule schedule) {
+                this.schedule = schedule;
+                return this;
+            }
+            public Builder withMinAppVersion(Integer minAppVersion) {
+                this.minAppVersion = minAppVersion;
+                return this;
+            }
+            public Builder withMaxAppVersion(Integer maxAppVersion) {
+                this.maxAppVersion = maxAppVersion;
+                return this;
+            }
+            public Builder addRequiredGroup(String... groups) {
+                for (String group : groups) {
+                    this.allOfGroups.add(group);
+                }
+                return this;
+            }
+            public Builder addProhibitedGroup(String... groups) {
+                for (String group : groups) {
+                    this.noneOfGroups.add(group);    
+                }
+                return this;
+            }
+            public ScheduleCriteria build() {
+                checkNotNull(schedule);
+                return new ScheduleCriteria(schedule, minAppVersion, maxAppVersion, allOfGroups, noneOfGroups);
+            }
```

Comment:
Yeah, see above. I've removed that check and now they should get InvalidEntityException indicating the schedule is missing.